### ISSUE TITLE
LitElement compatibility issues

### ISSUE THREAD

**Issue by rfool** [calebdwilliams/element-internals-polyfill#33]
`createHiddenInput` tests if the element has a property `updateComplete`, and if that exists it will insert the hidden input when that promise resolves:

```js
    const createHiddenInput = (ref, internals) => {
        const input = document.createElement('input');
        input.type = 'hidden';
        input.name = ref.getAttribute('name');
        if (ref.updateComplete) {
            ref.updateComplete.then(() => ref.after(input));
        }
        else {
            ref.after(input);
        }
        hiddenInputMap.get(internals).push(input);
        return input;
    };
```

Looks like that was written with LitElement in mind - which has that `updateComplete` property.

But it introduces two new issues when `setFormValue()` is called in `firstUpdated()` or `updated()`:

1. multiple calls to `setFormValue()` will result in multiple hidden inputs added,
2. actual form-association of element only comes into effect after `updateComplete` promise resolves.

I wonder why the `if( ref.updateComplete )` case was added, is it really required?

I'd suggest to remove that case, and always add the input immediately. That would resolve both issues.

**Comment by calebdwilliams** [comment]
Thanks for the issue. Could you create a reproduction (CodePen would be great) just so I can work from the right starting point?

**Comment by calebdwilliams** [comment]
Looks like this was introduced by @brion-fuller in #27. I'm wondering if any additional feedback could be provided here.

**Comment by brion-fuller** [comment]
Yeah the problem with in this case was how lit rendered elements asynchronously.  I will have to check for example but what would happen is the hidden element would effect static lit templates.  So rendering for the element would be broken.

**Comment by calebdwilliams** [comment]
Could you provide a bit more context around what you mean, @brion-fuller? I have [an example working on CodePen](https://codepen.io/calebdwilliams/pen/MWmrgLW?editors=0010), but don't want to introduce a regression for your issue.

**Comment by brion-fuller** [comment]
Is the CodePen with the removal of the `ref. updateComplete`.  It looks to be working for select elements.  
So the case I saw was mostly around select elements.

If you had something like 
```
html`
<select name="test">
    <option value="0">Hello</option>
    <option value="1">World</option>
    <option value="2">Foo</option>
    <option value="3">Bar</option>
</select>
`
This would cause the options to render as a string on the page like "Hello,World,Foo,Bar".

It looks like this works modifying your example.  
https://codepen.io/inbreakable/pen/XWRVNYJ?editors=0011

**Comment by calebdwilliams** [comment]
Perfect. I'll cut a new release. Thanks @brion-fuller and @rfool.

**Comment by calebdwilliams** [comment]
This should be fixed in `element-internals-polyfill-0.1.42`.
